c/tskit/tables.h

```diff
@@ -2733,6 +2767,48 @@ int tsk_table_collection_link_ancestors(tsk_table_collection_t *self, tsk_id_t *
     tsk_size_t num_samples, tsk_id_t *ancestors, tsk_size_t num_ancestors,
     tsk_flags_t options, tsk_edge_table_t *result);
 
+/* This is the current IBD C interface */
+int tsk_ibd_finder_free(tsk_ibd_finder_t *self);
+int tsk_ibd_finder_init_and_run(tsk_ibd_finder_t *ibd_finder,
+    tsk_table_collection_t *tables, double sequence_length, tsk_id_t *samples,
+    tsk_size_t num_samples, double min_length, double max_time);
+int tsk_ibd_finder_run(tsk_ibd_finder_t *ibd_finder);
+int tsk_ibd_finder_calculate_ibd(tsk_ibd_finder_t *self, tsk_id_t current_parent);
+void ibd_finder_print_state(tsk_ibd_finder_t *self, FILE *out);
+
+/* JK Here's the interface we want:
```

Comment:
I think we can simplify things quite a bit by changing the public interface. I've made a sketch of what this might look like here.

Reply:
I now have all the functions you've suggested, but have retained the `ibd_finder_init_and_run` to streamline the code inside the C tests. We don't need to use it in the interface itself.